### PR TITLE
rename JobStatus.Unsuccessful => JobStatus.Failed

### PR DESCRIPTION
**Pull request by mikedanese**

### CODE REVIEW

File: pkg/apis/experimental/validation/validation.go [hunk 1]
```diff
@@ -346,7 +346,7 @@ func ValidateJobStatus(status *experimental.JobStatus) errs.ValidationErrorList
 	allErrs := errs.ValidationErrorList{}
 	allErrs = append(allErrs, apivalidation.ValidatePositiveField(int64(status.Active), "active")...)
 	allErrs = append(allErrs, apivalidation.ValidatePositiveField(int64(status.Successful), "successful")...)
-	allErrs = append(allErrs, apivalidation.ValidatePositiveField(int64(status.Unsuccessful), "unsuccessful")...)
+	allErrs = append(allErrs, apivalidation.ValidatePositiveField(int64(status.Failed), "unsuccessful")...)
```
Comment (davidopp) [inline pkg/apis/experimental/validation/validation.go]: s/unsuccessful/failed/

File: pkg/controller/job/controller.go [hunk 2]
```diff
@@ -353,13 +353,13 @@ func newCondition() experimental.JobCondition {
...
-func getStatus(pods []api.Pod) (successful, unsuccessful int) {
+func getStatus(pods []api.Pod) (successful, failed int) {
 	successful = filterPods(pods, api.PodSucceeded)
-	unsuccessful = filterPods(pods, api.PodFailed)
+	failed = filterPods(pods, api.PodFailed)
 	return
 }
 
-func (jm *JobController) manageJob(activePods []*api.Pod, successful, unsuccessful int, job *experimental.Job) int {
+func (jm *JobController) manageJob(activePods []*api.Pod, successful, failed int, job *experimental.Job) int {
```
Comment (davidopp) [inline pkg/controller/job/controller.go]: I guess "failed" is unused in this function, but probably fine to keep it (I'm surprised Go doesn't complain?)

Comment (derekwaynecarr) [inline pkg/controller/job/controller.go]: @kubernetes/rh-cluster-infra

**Comment by derekwaynecarr** [comment]
Can this be cherry picked for 1.1?

File: pkg/controller/job/controller.go (hunk 2 above)
Comment (mikedanese) [inline pkg/controller/job/controller.go]: This is a remnant from when we treated RestartOnFailure differently then RestartNever. Should be okay to remove

**Comment by mikedanese** [comment]
@derekwaynecarr will do[

@davidopp addressed comments.

**Comment by davidopp** [comment]
LGTM

**Comment by derekwaynecarr** [comment]
Is there a reason why we would not rename Successful to Succeeded so things are consistent fully with the pod phases in tense?

**Comment by derekwaynecarr** [comment]
I think if we are changing this field we should bite the bullet and change both.

**Comment by derekwaynecarr** [comment]
Removing Lgtm pending response

**Comment by mikedanese** [comment]
@derekwaynecarr good catch. Fixed.

**Comment by davidopp** [comment]
LGTM

Appears to address Derek's concerns

**Comment by derekwaynecarr** [comment]
LGTM, thanks

**Comment by soltysh** [comment]
It's nice to see we got back to the naming from my initial proposal :wink:

**Pull request merged**